src/commons/controlBar/ControlBarShareButton.tsx

```diff
@@ -66,12 +89,36 @@ export class ControlBarShareButton extends React.PureComponent<ControlBarShareBu
                   onChange={this.handleChange}
                   style={{ width: 175 }}
                 />
+                <>{console.log(this.props.programConfig)}</>
                 <ControlButton
                   label="Get Link"
                   icon={IconNames.SHARE}
                   onClick={() => {
-                    this.props.handleShortenURL(this.state.keyword);
-                    this.setState({ isLoading: true });
+                    // post request to backend, set keyword as return uuid
+                    const requestBody = {
+                      shared_program: {
+                        data: this.props.programConfig
+                      }
+                    };
+                    const fetchOpts: RequestInit = {
+                      method: 'POST',
+                      body: JSON.stringify(requestBody),
+                      headers: {
+                        'Content-Type': 'application/json'
+                      }
+                    };
+                    fetch('http://localhost:4000/api/shared_programs', fetchOpts)
+                      .then(res => {
+                        return res.json();
+                      })
+                      .then(resp => {
+                        this.setState({
+                          keyword: 'http://localhost:8000/playground/share/' + resp.uuid
```

Comment:
Ditto on hardcoding.

Reply:
Hi, I didn't find any Constants of frontend URL, maybe can add one in the .env file?